packages/react-server/core/ReactServerAgent/Request.js

```diff
@@ -233,7 +236,37 @@ function buildSuperagentRequest() {
 		}
 	}
 
-	req.send(postParams);
+	// query parameters exist if there are more than one set
+	// post parameters exist if the value is not null
+	const hasQueryParams = (this._queryParams.length > 0),
+		hasPostParams = (this._postParams !== null);
+
+	if (hasPostParams) {
+		// superagent has some weird, implicit file upload support
+		// that only works if you don't set `type`.
+		if (this._type && this._type !== 'form-data') {
+			req.type(this._type);
+		}
+
+		req.send(postParams);
+	}
+
+	switch (this._method) {
+		case 'GET':
+		case 'HEAD':
+			if (hasPostParams) {
+				logger.warning(`Attempting a ${this._method} request with POST data (using SuperAgent's .send() function). This might result in a CORS preflight request`);
+			}
+			break;
+
+		case 'POST':
+		case 'PATCH':
+		case 'PUT':
+			if (hasQueryParams) {
+				logger.info(`Attempting a ${this._method} request with query data (using SuperAgent's .query() function). This might not be what you want.`);
+			}
+			break;
```

Comment:
I don't think so.  If someone is using a method other than GET, HEAD, POST, PUT, or PATCH and they're setting query/post parameters, then they probably know what they are doing and aren't worried about a CORS preflight request being emitted.